lua/treesitter-context/render.lua

```diff
@@ -20,15 +22,43 @@ local retired_buffers = {}
 local window_contexts = {}
 
 --- @return integer buf
-local function create_buf()
+local function create_or_get_buf()
+  for index=#buffers_pool, 1, -1 do
+    local buf = buffers_pool[index]
+    if api.nvim_buf_is_valid(buf) then
+      table.remove(buffers_pool, index)
+      return buf
+    end
+  end
+
   local buf = api.nvim_create_buf(false, true)
 
   vim.bo[buf].undolevels = -1
-  vim.bo[buf].bufhidden = 'wipe'
```

Comment:
This option deletes the buffer after the window is closed. It appears that the problem existed before multi-window support was introduced.